pkgs/build-support/cc-wrapper/cc-wrapper.sh

```diff
@@ -180,7 +180,18 @@ fi
 
 PATH="$path_backup"
 # Old bash workaround, see above.
-exec @prog@ \
-    ${extraBefore+"${extraBefore[@]}"} \
-    ${params+"${params[@]}"} \
-    ${extraAfter+"${extraAfter[@]}"}
+
+if (( "${NIX_CC_USE_RESPONSE_FILE:-@use_response_file_by_default@}" >= 1 )); then
+    RESPONSE_FILE=$(mktemp)
+    printf "%q\n" \
+       ${extraBefore+"${extraBefore[@]}"} \
+       ${params+"${params[@]}"} \
+       ${extraAfter+"${extraAfter[@]}"} \
+       > "${RESPONSE_FILE}"
+    exec @prog@ "@${RESPONSE_FILE}"
```

Comment:
```suggestion
    exec @prog@ "@<(printf "%q\n" \
       ${extraBefore+"${extraBefore[@]}"} \
       ${params+"${params[@]}"} \
       ${extraAfter+"${extraAfter[@]}"})"
```

Reply:
TIL about `<()` (and `=()`). Thanks. This looks good.